fw/Core/Hitcon/Logic/Display/test-display.cc

```diff
@@ -71,8 +81,20 @@ int main() {
         0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0,
         // clang-format on
     };
+    constexpr int scroll_n_col =
+        sizeof(buf_scroll) / (sizeof(uint8_t) / sizeof(char));
+    uint8_t buf_scroll_compressed[scroll_n_col];
+    for (int i = 0; i < DISPLAY_HEIGHT; i++) {
+      for (int j = 0; j < scroll_n_col; j++) {
+        if (buf_scroll[i * scroll_n_col + j])
+          DISPLAY_SET(buf_scroll_compressed[j], i);
+        else
+          DISPLAY_UNSET(buf_scroll_compressed[j], i);
+      }
+    }
```

Comment:
This code seems to be duplicated. Maybe create a function (or macro) to compress display buffer from 2d to 1d?